plugins/robothor_plugin/robothor_distributions.py

```diff
@@ -0,0 +1,41 @@
+from typing import Tuple, Union, cast
+
+import torch
+
+from core.base_abstractions.distributions import CategoricalDistr, Distr
+
+
+class TupleCategoricalDistr(Distr):
+    def __init__(self, probs=None, logits=None, validate_args=None):
+        if probs is not None:
+            params = [
+                CategoricalDistr(probs=probs[..., it, :], validate_args=validate_args)
+                for it in range(probs.shape[-2])
+            ]
+        else:
+            params = [
+                CategoricalDistr(logits=logits[..., it, :], validate_args=validate_args)
+                for it in range(logits.shape[-2])
+            ]
```

Comment:
Not sure how much this matters, but it is probably more efficient to create a single `CategorialDistr`, e.g.
```python
CategoricalDistr(probs=probs.view(-1, probs.shape[-1]), validate_args=validate_args)
```
and then, when computing the below functions, just reshape after computing things.

Reply:
That was actually my initial implementation :) I was just trying different things and this one is the one that settled, but I can move back to what you propose.

Reply:
If the speed is similar then I don't mind either :) I think the way you have it is actually probably more intuitive but we might save some speed with the vectorization.

Reply:
I went to a design where we have a single distribution. It's not much cleaner in code, except for the initializer, but it might be a little faster 👍